### ISSUE TITLE
tsh logout deletes the current-profile and yaml configuration

### ISSUE THREAD

**Issue by lukavia** [gravitational/teleport#12148]
### Description

**What happened**:
tsh would delete all files from .tsh including the current-profile and proxy.yaml files that would be needed on next login. This forces the use of --proxy and --user flag on next login again. That is in contrast with previous version and with the behaviour if your certificate just have expired. 

**What you expected to happen**:
Once I use tsh login --proxy xxx --user yyy and then logout again I would expect the next time to just use tsh login.

### Reproduction Steps
As minimally and precisely as possible, describe step-by-step how to reproduce the problem.
1. tsh login --proxy xxx.example.com --user myuser
2. tsh logout
3. tsh login
ERROR: No proxy address specified, missed --proxy flag?

### Server Details
- Teleport version (run `teleport version`): Teleport v9.0.4 git:v9.0.4-0-gf577413d3 go1.17.7
- Server OS (e.g. from `/etc/os-release`): Debian GNU/Linux 10 (buster) 
- Where are you running Teleport? (e.g. AWS, GCP, Dedicated Hardware): Dedicated Hardware

### Client Details
- Tsh version (`tsh version`):Teleport v9.0.4 git:v9.0.4-0-gf577413d3 go1.17.7
- Computer OS (e.g. Linux, macOS, Windows): Linux & Windows
- Installed via (e.g. apt, yum, brew, website download): apt and website download

**Comment by zmb3** [comment]
@atburke is this related to the new config file for tsh?

**Comment by atburke** [comment]
I don't think I've worked on anything related to tsh config files, so I don't know

**Comment by zmb3** [comment]
Whoops, my bad - that was @lxea. Alex - any ideas on this one?

**Comment by lxea** [comment]
I dont think this is related to the .tshconfig changes, I have the same behavior with commit 4f09a8ade31d50903a2dc98fb131bfa9097962ee (which is 1 commit before .tsh config got merged) as well as after

**Comment by GavinFrazar** [comment]
This only seems to happen if I do `$ tsh logout` (logout from all users). If I specify proxy and user, it does not delete everything: `$ tsh logout --proxy=proxy.example.local --user=gavin`.
